src/opendr/perception/panoptic_segmentation/efficient_lps/efficient_lps_learner.py

```diff
@@ -0,0 +1,829 @@
+        h = loader_pipeline_cfg["H"]
+        w = loader_pipeline_cfg["W"]
+        fov_up = loader_pipeline_cfg["fov_up"]
+        fov_down = loader_pipeline_cfg["fov_down"]
+        max_points = loader_pipeline_cfg["max_points"]
+        sensor_img_means = loader_pipeline_cfg["sensor_img_means"]
+        sensor_img_stds = loader_pipeline_cfg["sensor_img_means"]
+
+        scan = LaserScan(project=project, H=h, W=w, fov_up=fov_up, fov_down=fov_down)
+        remissions = None
+        shape = point_cloud.data.shape
+        if shape[-1] < 3:
+            raise ValueError("Invalid point cloud shape")
+        elif shape[-1] == 3:
+            points = point_cloud.data
+        else:
+            points = point_cloud.data[:, :3]
+            remissions = point_cloud.data[:, 3]
+
+        scan.set_points(points, remissions=remissions)
+        scan.do_range_projection()
+
+        unproj_n_points = scan.points.shape[0]
+        unproj_xyz = torch.full((max_points, 3), -1.0, dtype=torch.float)
+        unproj_xyz[:unproj_n_points] = torch.from_numpy(scan.points)
+        unproj_range = torch.full([max_points], -1.0, dtype=torch.float)
+        unproj_range[:unproj_n_points] = torch.from_numpy(scan.unproj_range)
+        unproj_remissions = torch.full([max_points], -1.0, dtype=torch.float)
+        unproj_remissions[:unproj_n_points] = torch.from_numpy(scan.remissions)
+
+        sensor_img_means = torch.tensor(sensor_img_means,
+                                        dtype=torch.float).view(-1, 1, 1)
+        sensor_img_stds = torch.tensor(sensor_img_stds,
+                                       dtype=torch.float).view(-1, 1, 1)
+
+        # get points and labels
+        proj_range = torch.from_numpy(scan.proj_range).clone()
+        proj_xyz = torch.from_numpy(scan.proj_xyz).clone()
+        proj_remission = torch.from_numpy(scan.proj_remission).clone()
+        proj_mask = torch.from_numpy(scan.proj_mask)
+        proj_x = torch.full([max_points], -1, dtype=torch.long)
+        proj_x[:unproj_n_points] = torch.from_numpy(scan.proj_x)
+        proj_y = torch.full([max_points], -1, dtype=torch.long)
+        proj_y[:unproj_n_points] = torch.from_numpy(scan.proj_y)
+        proj = torch.cat([proj_range.unsqueeze(0).clone(),
+                          proj_xyz.clone().permute(2, 0, 1),
+                          proj_remission.unsqueeze(0).clone()])
+        proj = (proj - sensor_img_means
+                ) / sensor_img_stds
+        proj = proj * proj_mask.float()
+
+        mmdet_pcl = {
+            # TODO: Improve Awful Hack!
+            # Prediction will be saved in $PWD/tmpDir/inference/predictions/pcl_<file_count>.label
+            #                                         |||||||||             \\\\\\\\\\\\\\\\\
+            #                                         VVVVVVVVV              VVVVVVVVVVVVVVVVV
+            "filename": f"expected_path_for_something/inference/__dont_care__/pcl_{frame_id:06d}.bin",
```

Comment:
can it be improved? :smile:

Reply:
I have tried to improve this by changing the original repository. Now the filename is not used at all since the 3D predictions are not saved/loaded locally but instead values are passed as variables directly. I think this should do the trick